src/libraries/System.Private.CoreLib/src/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -218,6 +220,12 @@ public override void Return(T[] array, bool clearArray = false)
             if (log.IsEnabled())
             {
                 log.BufferReturned(array.GetHashCode(), array.Length, Id);
+                if (!(haveBucket & returned))
```

Comment:
perhaps we should have a style preference for `&` vs `&&` when both sides are booleans. It seems it makes the same IL. I always use `&&`.